### DISCUSSION TITLE
how to upsert 1-1 with both as primary key?

### DISCUSSION THREAD

**Discussion by Felonius** [prisma/prisma#5776] [Q&A]
HI everyone, 
new to Prisma.  running Prisma 2 and have imported from an existing app's data source.  able to introspect and generate the client:

```prisma
model FileData {
  id       Int      @id
  fileData Bytes?
  FileMeta FileMeta @relation(fields: [id], references: [id])
}

model FileMeta {
  id       Int       @id @default(autoincrement())
  title    String?   @db.VarChar(150)
  fileName String?   @db.VarChar(150)
  fileType String?   @db.VarChar(50)
  byteSize String?   @db.VarChar(50)
  isThumb  Boolean?  @db.Bit(1)
  FileData FileData?
}
```

how do I upset records to these?  can't figure it out.

**Comment by ryands17** [comment]
@Felonius 👋 

Do you want to create a relation data? If so you can use connect as follows:

```ts
await prisma.fileData.create({
    data: {
      fileData: Buffer.from('data'),
      FileMeta: { create: { fileName: 'name', ...rest of the fields } },
    },
  })
```

You can read more about this [here](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#nested-writes). Let me know if this works :)

**Reply by Felonius** [reply]
Hi @ryands17 !  thanks very much for responding. this looks promising. I suspect I keep getting this confused. I feel like the declaration is backwards.  its not working at the moment but I've messed up my code a bit so I'll need to play with it tomorrow.  thanks for your help!

**Reply by ryands17** [reply]
Let me know if it works 👍

**Reply by Felonius** [reply]
Hey @ryands17 , sorry for not responding sooner.  I'm not sure if this works or not sorry.  this is the create portion of the code, however I'm finding my update runs through updating an existing record so I'm doing an upsert inside an update, which works:

[previously stated fileDatafields]

```ts
FileMeta.FileData = {
        upsert: {
          create: fileDataFields,
          update: fileDataFields,
        }
      }

let retFileMeta = await prisma.fileMeta.update({
        where: { id: FileMeta.id },
        data: FileMeta,
        include: { FileData: true, },
      });
```

**Reply by ryands17** [reply]
@Felonius 👋 

I still am unable to get what part is not working. Do you want to create `FileMeta` and `FileData` together or do you want to add `FileData` to a record in `FileMeta`?

**Reply by Felonius** [reply]
heya @ryands17 .  its the structure of the UPSERT I needed. You'd provided a create. I can't expect just because I have a FileMeta record that the FileData would be there or vice versa, so an upsert is needed.  the code I responded with does that.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
